### ISSUE TITLE
Segfault with matrix multiply on 0 by 0 matrix

### ISSUE THREAD

**Issue by eriknw** [DrTimothyAldenDavis/GraphBLAS#112]
We are seeing segfaults when doing matrix multiply (mxv or mxm) on matrices with shape `(0, 0)`.

For example,
```C
  GrB_Matrix_new(&M_0, GrB_INT64, 0, 0);
  GrB_Matrix_assign_INT64(M_0, NULL, NULL, 1, GrB_ALL, 0, GrB_ALL, 0, NULL);
  GrB_Vector_new(&v_0, GrB_INT64, 0);
  GrB_Vector_assign_INT64(v_0, NULL, NULL, 1, GrB_ALL, 0, NULL);
  GrB_Vector_new(&v_1, GrB_INT64, 0);
  GrB_mxv(v_1, NULL, NULL, GrB_PLUS_TIMES_SEMIRING_INT64, M_0, v_0, NULL);
```
Note that to see the segfault, it's necessary to perform assignments as above.

I see this for the following semirings (use bool or uint datatypes where appropriate):
```
bxnor_band
bxnor_bor
bxnor_bxnor
bxnor_bxor
bxor_band
bxor_bor
bxor_bxnor
bxor_bxor
eq_eq
eq_first
eq_ge
eq_gt
eq_land
eq_le
eq_lor
eq_lt
eq_lxor
eq_ne
eq_second
lxnor_lor
lxor_eq
lxor_first
lxor_ge
lxor_gt
lxor_land
lxor_le
lxor_lor
lxor_lt
lxor_lxor
lxor_ne
lxor_pair
lxor_second
plus_div
plus_first
plus_iseq
plus_isge
plus_isgt
plus_isle
plus_islt
plus_isne
plus_land
plus_lor
plus_lxor
plus_max
plus_min
plus_minus
plus_pair
plus_plus
plus_rdiv
plus_rminus
plus_second
plus_times
times_div
times_first
times_iseq
times_isge
times_isgt
times_isle
times_islt
times_isne
times_land
times_lor
times_lxor
times_max
times_min
times_minus
times_plus
times_rdiv
times_rminus
times_second
times_times
```
Credit to @ParticularMiner for first spotting this and reporting it here: https://github.com/metagraph-dev/grblas/issues/168

**Comment by DrTimothyAldenDavis** [comment]
Got it.  Yes, that's definitely a (gasp!) SuiteSparse:GraphBLAS bug.  Stack overflow in Source/GB_iso_reduce_worker.c, when n is zero.  The test should be "if (n <= 1)...".  If n is zero then the reduction gets stuck in an infinite loop and the stack overflows.  See https://github.com/DrTimothyAldenDavis/GraphBLAS/commit/1428379dfe045fb1d91ac268195889bd97f1c12e .  This makes the segfault go away.  I have to verify that this is the right fix, and do all my tests.

**Comment by DrTimothyAldenDavis** [comment]
Fixed in v6.2.4.
